src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs

```diff
@@ -509,6 +516,8 @@ await InitializeConnectionAsync(async context =>
         appDelegateTcs = new TaskCompletionSource();
         await StartStreamAsync(1, _browserRequestHeaders, endStream: true);
 
+        // If app code is running we know the stream has been created
+        await appTcs.Task;
```

Comment:
On line 544, a new stream is created: `await StartStreamAsync(3, _browserRequestHeaders, endStream: true);`

The test should create a new TCS instance, assign to appTcs and await again before looking at the stream pool.